pkg/util/workloads_utils.go

```diff
@@ -221,6 +324,26 @@ func ParseReplicasFrom(object *unstructured.Unstructured) int32 {
 	return replicas
 }
 
+func ParsePodTemplate(object *unstructured.Unstructured) *v1.PodTemplateSpec {
+	t, found, err := unstructured.NestedFieldNoCopy(object.Object, "spec", "template")
+	if err != nil || !found {
+		return nil
+	}
+	template := &v1.PodTemplateSpec{}
+	templateByte, _ := json.Marshal(t)
+	_ = json.Unmarshal(templateByte, template)
+	return template
+}
+
+func ParseInt32PartitionFrom(object *unstructured.Unstructured) int32 {
+	partition := int32(0)
+	field, found, err := unstructured.NestedInt64(object.Object, "spec", "updateStrategy", "rollingUpdate", "partition")
```

Comment:
This should not be a common logic in workload util, for it is only valid for statefulset-like workloads.